### PR TITLE
Use lambda and functional notation

### LINKED ISSUE #103: Use functional notation for runnables/callables/threads
style issue / java8

### PR DESCRIPTION
**Pull request by amcp**
Fixes #103

### CODE REVIEW

**Comment by vinaykumarchella** [comment]
I will get to it next week

**Comment by amcp** [comment]
@vinaykumarchella ready for review

**Comment by arunagrawal84** [comment]
LGTM

File: ndbench-core/src/main/java/com/netflix/ndbench/core/NdBenchDriver.java [hunk 1]
```diff
@@ -103,19 +101,16 @@
...
-                    Thread.sleep(2000);
-                } catch (Exception e) {
-                    //ignore
-                }
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            logger.info("*** shutting down NdBench server since JVM is shutting down");
+            NdBenchDriver.this.stop();
+            try {
+                NdBenchDriver.this.shutdownClient();
+                Thread.sleep(2000);
```
Comment (ipapapa) [inline ndbench-core/src/main/java/com/netflix/ndbench/core/NdBenchDriver.java]: if the JVM is shutting down, do we have enough time to sleep here?

Comment (vinaykumarchella) [reply ndbench-core/src/main/java/com/netflix/ndbench/core/NdBenchDriver.java]: This is the best effort, does not need to wait

**Pull request merged**